fitbenchmarking/results_processing/performance_profiler.py

```diff
@@ -0,0 +1,197 @@
+            # if we don't need a log plot, then don't print one
+            fig, ax = plt.subplots(1, 2,
+                                   gridspec_kw={
+                                       'width_ratios': [30, 7.8],
+                                       'wspace': 0.01})
+            legend_ax = 1
+            use_log_plot = False
+        else:
+            fig, ax = plt.subplots(1, 3,
+                                   sharey=True,
+                                   gridspec_kw={
+                                       'width_ratios': [10, 20, 7.8],
+                                       'wspace': 0.01})
+            legend_ax = 2
+
+        # Plot linear performance profile
+        create_plot(ax[0], step_values, acc.keys())
+        ax[0].set_xlim(1, linear_upper_limit)
+        ax[0].set_xticks([1, 2, 4, 6, 8, 10])
+        ax[0].set_xticklabels(['$1$', '$2$', '$4$', '$6$', '$8$', '$10$'])
+        ax[0].yaxis.set_ticks_position('left')
+
+        if use_log_plot:
+            # Plot log performance profile
+            create_plot(ax[1], step_values, acc.keys())
+            ax[1].set_xlim(
+                linear_upper_limit,
+                min(max_value+1, 10000))
+            ax[1].set_xscale('log')
+            ax[1].set_xticks([100, 1000, 10000])
+            ax[1].set_xticklabels(['$10^2$', '$10^3$', '$10^4$'])
+            ax[1].yaxis.set_ticks_position('right')
+            ax[1].tick_params(axis='y', labelcolor='white')
+
+        # legend
+        ax[legend_ax].axis('off')
+        handles, labels = ax[0].get_legend_handles_labels()
+        wrapped_labels = ['\n'.join(wrap(l, 22)) for l in labels]
+        ax[legend_ax].legend(handles, wrapped_labels, loc=2, prop={'size': 7})
+
+        # Common parts
+        plt.ylim(0.0, 1.0)
+        fig.suptitle("Performance profile - {}".format(name))
+
+        # add a big axis, hide frame
+        fig.add_subplot(111, frameon=False)
+        # hide tick and tick label of the big axis
+        plt.tick_params(labelcolor='none',
+                        top=False,
+                        bottom=False,
+                        left=False,
+                        right=False)
+        # the xlabel has added white space to (empirically) make it look
+        # centred under the plots (and not the legend)
+        plt.xlabel("f                       ")
+        plt.ylabel("fraction for which solver within f of best")
+        ax[0].set_ylim(0.0, 1.05)
+
+        plt.savefig(this_filename, dpi=150)
+
+    return figure_path
+
+def create_plot(ax, step_values, keys):
+    """
+    Function to draw the profile on a matplotlib axis
+
+    :param ax : A matplotlib axis to be filled
+    :type ax :  an `.axes.SubplotBase` subclass of `~.axes.Axes`
+                (or a subclass of `~.axes.Axes`)
+    :param step_values : a sorted list of the values of the metric
+                         being profiled
+    :type step_values : list of float
+    """
```

Comment:
Keys is missing from the docstring

Reply:
Fixed.  Also changed from passing `keys` to passing a list of labels, as we don't really need the keys at all.